### PR TITLE
WL-3119 Ensure a event creator is defined

### PR DESCRIPTION
**Pull request by ghost**
This ensures that an attendee gets assigned to the event, if this doesn't
happen the calendar will be invalid because it uses the `method=REQUEST`.

Infuriatingly the setCreator method on an event takes no parameters and
will assign the current user. This is not necessarily the creator of the
event.

Fortunately the event exposes its properties so we can just fiddle with
them directly. What could go wrong?

### CODE REVIEW

**Comment by buckett** [comment]
This seems to put the person using the UI at the time that the event was created as the CHAIR of the iCal event. It would seem that it should be the organisers who are listed as the CHAIRs of the iCal event?

It strikes me that the field organiser is mis-labeled in the Signup tool UI as when I hear that I think of the secretary organising a set of signups on behalf of a lecturer, but I think that the signup tool actually means chairman when it talks about organiser.

**Comment by ghost** [comment]
I've tried to be good with my commits, they should tell you everything you need to know.

File: api/src/java/org/sakaiproject/signup/logic/SignupCalendarHelper.java [hunk 1]
```diff
@@ -54,6 +54,14 @@
 	public String createCalendarFile(List<ExtEvent> vevents);
 	
 	/**
+	 * Create a calendar for a list of ExtEvents and return the path to the file
+	 * @param vevents	List of ExtEvents
+	 * @param method	The ITIP method for the calendar, e.g. "REQUEST"
+	 * @return
```
Comment (buckett) [inline api/src/java/org/sakaiproject/signup/logic/SignupCalendarHelper.java]: Missing return param doc.

Comment (ghost) [inline api/src/java/org/sakaiproject/signup/logic/SignupCalendarHelper.java]: Done

File: impl/src/java/org/sakaiproject/signup/logic/SignupCalendarHelperImpl.java [hunk 2]
```diff
@@ -245,4 +280,7 @@ private String getSiteAccessUrl(String siteId) {
 	@Setter
 	private ExternalCalendaringService externalCalendaringService;
 
+	@Setter
```
Comment (buckett) [inline impl/src/java/org/sakaiproject/signup/logic/SignupCalendarHelperImpl.java]: The idea with the SakaiFacade is that all Sakai related code lives there and so the signup tool only talks to the SakaiFacade.

The loading of users should be done in the SakaiFacade.

Comment (ghost) [inline impl/src/java/org/sakaiproject/signup/logic/SignupCalendarHelperImpl.java]: Good point, mercifully the `SakaiFacade` already has exactly that method. :+1:

File: impl/src/java/org/sakaiproject/signup/logic/SignupCalendarHelperImpl.java [hunk 3]
```diff
@@ -137,7 +143,27 @@ public ExtEvent generateExtEventForMeeting(SignupMeeting meeting) {
...
+	 * @param meeting  the meeting in question
+	 * @return the list of coordinator Users
+	 */
+	private List<User> getCoordinators(SignupMeeting meeting) {
+		List<User> users = new ArrayList<User>();
+		List<String> ids = meeting.getCoordinatorIdsList();
+		for (String coordinator : ids) {
+			try {
+				users.add(sakaiFacade.getUser(coordinator));
+			} catch (UserNotDefinedException e) {
```
Comment (buckett) [inline impl/src/java/org/sakaiproject/signup/logic/SignupCalendarHelperImpl.java]: I don't think it throws UserNotDefinedException. I'd also be tempted to use getUserQuietly as if a coordinator has been deleted from Sakai there's no point in logging it all over the place.

Comment (ghost) [inline impl/src/java/org/sakaiproject/signup/logic/SignupCalendarHelperImpl.java]: Good point. The resulting code is much cleaner. Rewritten.

**Pull request merged**